src/geometry/utility/geometry.cc

```diff
@@ -0,0 +1,308 @@
+using Segment2d = std::pair<maliput::math::Vector2, maliput::math::Vector2>;
+
+namespace {
+
+// Determines whether two line segments intersects.
+//
+// Based on https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line_segment
+bool SegmentsIntersect2d(const Segment2d& lhs, const Segment2d& rhs) {
+  const auto& x1 = lhs.first[0];
+  const auto& y1 = lhs.first[1];
+  const auto& x2 = lhs.second[0];
+  const auto& y2 = lhs.second[1];
+  const auto& x3 = rhs.first[0];
+  const auto& y3 = rhs.first[1];
+  const auto& x4 = rhs.second[0];
+  const auto& y4 = rhs.second[1];
+  const double den = (x1 - x2) * (y3 - y4) - (y1 - y2) * (x3 - x4);
+  if (den == 0) {
+    // They are parallel or coincident.
+    return false;
+  }
+  const double t_num = (x1 - x3) * (y3 - y4) - (y1 - y3) * (x3 - x4);
+  const double u_num = (x1 - x3) * (y1 - y2) - (y1 - y3) * (x1 - x2);
+  const double t = t_num / den;
+  const double u = u_num / den;
+  return (t >= 0 && t <= 1 && u >= 0 && u <= 1);
+}
+
+// Obtains the distance between @p lhs and @p rhs points projected on the xy plane.
+double Distance2d(const maliput::math::Vector3& lhs, const maliput::math::Vector3& rhs) {
+  const maliput::math::Vector2 lhs_2d{lhs.x(), lhs.y()};
+  const maliput::math::Vector2 rhs_2d{rhs.x(), rhs.y()};
+  return (lhs_2d - rhs_2d).norm();
+}
+
+// Determines whether point @p p is located left from segment compound by @p seg_first and @p seg_second.
+bool PointIsLeftOf(const maliput::math::Vector3& seg_first, const maliput::math::Vector3& seg_second,
+                   const maliput::math::Vector3& p) {
+  return ((seg_second - seg_first).cross(p - seg_first)).z() > 0;
+}
+
+maliput::math::Vector3 MakeCenterpoint(const maliput::math::Vector3& p1, const maliput::math::Vector3& p2) {
+  return (p1 + p2) / 2.;
+}
+
+// Helper class for evaluating the boundaries of a lane.
+class BoundChecker {
+ public:
+  BoundChecker(const LineString3d& left, const LineString3d& right)
+      : left_(left), right_(right), entry_{right.first(), left.first()}, exit_{left.last(), right.last()} {
+    left_segments_ = MakeSegments(left_);
+    right_segments_ = MakeSegments(right_);
+  }
+
+  bool Intersects2d(const Segment3d& seg) const {
+    const Segment2d seg2d{{seg.first.x(), seg.first.y()}, {seg.second.x(), seg.second.y()}};
+    const bool result =
+        IntersectsLeft2d(seg2d) || IntersectsRight2d(seg2d) || CrossesEntry2d(seg2d) || CrossesExit2d(seg2d);
+    return result;
+  }
+
+  bool IntersectsLeft2d(const Segment2d& seg) const {
+    for (const auto left_segment : left_segments_) {
+      const Segment2d left_seg{{left_segment.first.x(), left_segment.first.y()},
+                               {left_segment.second.x(), left_segment.second.y()}};
+      if (SegmentsIntersect2d(seg, left_seg)) {
+        if (!(seg.first == left_seg.first)) {
+          return true;
+        }
+      }
+    }
+    return false;
```

Comment:
I know it might require the same lines of code but it could be self-explanatory if you convert this into a pipeline of:

std::transform -> std::find

Reply:
Done. Using a `std::find_if` instead.